Tests/powershellcore/U031.PSCoreScripts.Copy-SitefinityDatabase.Tests.ps1

```diff
@@ -0,0 +1,298 @@
+
+        Mock Get-AzWebApp -MockWith { return @{
+            Name          = 'dfc-foo-as'
+            Kind          = 'app'
+            ResourceGroup = 'dfc-foo-rg'
+            Type          = 'Microsoft.Web/sites'
+            Id            = '/subscriptions/mock-sub/resourceGroups/dfc-foo-rg/providersMicrosoft.Web/sites/dfc-foo-as'
+            SiteConfig    = @{ AppSettings = @(
+                @{ Name = 'DatabaseVersion'; Value = 'dfc-foo-sitefinitydb' }
+            ) }
+        } }
+    
+        Mock Get-AzSqlDatabase -ParameterFilter { $DatabaseName -eq "dfc-foo-sitefinitydb" } -MockWith { return @{
+            DatabaseName      = 'dfc-foo-sitefinitydb'
+            ServerName        = 'dfc-foo-sql'
+            ResourceGroupName = 'dfc-foo-rg'
+            ResourceId        = '/subscriptions/mock-sub/resourceGroups/dfc-foo-rg/providers/Microsoft.Sql/servers/dfc-foo-sql/databases/dfc-foo-sitefinitydb'
+            SkuName           = 'Standard'
+            ElasticPoolName   = $null
+        } }
+
+        Mock Get-AzSqlDatabase -ParameterFilter { $DatabaseName -eq "dfc-foo-sitefinitydb-r123" } -MockWith { return $null }
+
+        ./Copy-SitefinityDatabase -AppServiceName dfc-foo-as -ServerName dfc-foo-sql -ReleaseNumber 123
+
+        It "should get the sql server resource description" {
+            Assert-MockCalled Get-AzResource -Exactly 1 -ParameterFilter { $Name -eq "dfc-foo-sql" }
+        }
+
+        It "should get the web app resource description" {
+            Assert-MockCalled Get-AzResource -Exactly 1 -ParameterFilter { $Name -eq "dfc-foo-as" }
+        }
+
+        It "should get the web app details" {
+            Assert-MockCalled Get-AzWebApp -Exactly 1 -ParameterFilter {
+                $Name -eq "dfc-foo-as" -and `
+                $ResourceGroupName -eq "dfc-foo-rg"
+            }
+        }
+        
+        It "should get the existing table" {
+            Assert-MockCalled Get-AzSqlDatabase -Exactly 1 -ParameterFilter {
+                $ResourceGroupName -eq "dfc-foo-rg" -and `
+                $ServerName -eq "dfc-foo-sql" -and `
+                $DatabaseName -eq "dfc-foo-sitefinitydb"
+            }
+        }
+
+        It "should look for a copy database called dfc-foo-sitefinitydb-r123" {
+            Assert-MockCalled Get-AzSqlDatabase -Exactly 1 -ParameterFilter {
+                $ResourceGroupName -eq "dfc-foo-rg" -and `
+                $ServerName -eq "dfc-foo-sql" -and `
+                $DatabaseName -eq "dfc-foo-sitefinitydb-r123"
+            }
+        }
+
+        It "Should copy dfc-foo-sitefinitydb to dfc-foo-sitefinitydb-r123" {
+            Assert-MockCalled New-AzSqlDatabaseCopy -Exactly 1 -ParameterFilter {
+                $ResourceGroupName -eq "dfc-foo-rg" -and `
+                $ServerName -eq "dfc-foo-sql" -and `
+                $DatabaseName -eq "dfc-foo-sitefinitydb" -and `
+                $CopyDatabaseName -eq "dfc-foo-sitefinitydb-r123"
```

Comment:
Should add a parameter filter for the elastic pool being $null here?

Reply:
I wasn't sure if $null was the same as checking the parameter is not passed. The tests still pass if I check for $null so I have added this